etcd3/client.py

```diff
@@ -318,8 +318,10 @@ def get_prefix(self, key_prefix, **kwargs):
         :returns: sequence of (value, metadata) tuples
         """
         range_response = self.get_prefix_response(key_prefix, **kwargs)
-        for kv in range_response.kvs:
-            yield (kv.value, KVMetadata(kv, range_response.header))
+        return (
```

Comment:
this is a bit sneaky, but returning a generator comp instead of having this method be a generator function means the kwarg check happens straight away, instead of whenever someone actually calls `next()` on the response

this changes when the data is actually fetched too.
I'm happy to rework this with a wrapper around the generator function instead, or by fudging in the test if you'd rather leave it as-is

Reply:
(could be argued that the other generator functions should be modified to behave like this too)

Reply:
I think this is fine. If you're feeling like it, you could also update the other generator functions